src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,133 @@
+#include "depthai/depthai_wrapper.hpp"
+#include <fstream>
+#include <sstream>
+
+#define WARNING "\033[1;5;31m" // Adds color to error statements
+#define ENDC "\033[0m"
+
+namespace DepthAI {
+
+const std::unordered_map<int, int> DepthAI::width_to_height_map_ = { { 720, 1280 }, { 800, 1280 }, { 400, 640 } };
+
+DepthAI::DepthAI(const std::string& usb_device, const std::string& config_file, bool usb2_mode)
+    : Device(usb_device, usb2_mode)
+{
+    // opening config file
+    std::ifstream file(config_file);
+    std::ostringstream file_stream;
+    if (file) {
+        file_stream << file.rdbuf();
+    } else {
+        std::cerr << WARNING + config_file + " file  is not found" ENDC << std::endl;
```

Comment:
Umm this is not the standard way to raise runtime errors. You can just do the following 
```
throw std::runtime_error("Config file could not be found at " + config_file); 
```